package/MDAnalysis/auxiliary/xvg.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from . import base
+
+class XVGReader(base.AuxFileReader):
+    """ Read data from .xvg file
+    
+    Assumes data is time-ordered and first column is time
+
+    Currently reading from file; can probably load a once and read through array
+    instead
+    """
+ 
+    def __init__(self, auxname, filename, **kwargs):
+        super(XVGReader, self).__init__(auxname, time_first_col=True, **kwargs)
+        ## should generalise in case time not first column...
+        
+    def _read_next_step(self):
+        """ Read next recorded timepoint in xvg file """
+        line = self.auxfile.readline()
+        if line:
+            # xvg has both comments '#' and grace instructions '@'
+            while line.lstrip()[0] in ['#', '@']:

```

Comment:
A small **untested** filter to move the responsibility of removing comments (snippets from the top of my head, take with caution):

```python
def uncomment(lines):
    for line in lines:
        # '@' must be at the beginning of a line to be a grace instruction
        if line.lstrip()[0] == '@':
            continue
        # '#' can be anywhere in the line, everything after is a comment
        comment_position = line.find('#')
        if comment_position > 0 and line[:coment_position]:
             yield line[:coment_position]
        elif comment_position <0 and line:
            yield line
        # if comment_position == 0, then the line is empty
```